src/strings/ops.c

```diff
@@ -509,6 +518,13 @@ MVMString * MVM_string_repeat(MVMThreadContext *tc, MVMString *a, MVMint64 count
     if (agraphs == 0)
         return tc->instance->str_consts.empty;
 
+    /* Total size of the resulting string can't be bigger than an MVMString is allowed to be. */
+    total_graphs = (MVMuint64)agraphs * (MVMuint64)count;
```

Comment:
Since count is a 64-bit integer, it's possible that this bounds check itself could overflow.